enclave_manager/avalon_enclave_manager/avalon_enclave_info.py

```diff
@@ -267,3 +274,70 @@ def create_signup_info(nonce):
 
     # Now we can return the real object
     return signup_info_obj
+
+
+# -----------------------------------------------------------------
+class EnclaveInfo(object):
+    """
+    Class to create/hold signup information for an enclave which
+    includes keys, sealed_data, enclave_id,etc.
+    """
+
+    # -------------------------------------------------------
+    @classmethod
+    def create_enclave_signup_data(cls):
+        """
+        Create enclave signup data
+        """
+
+        nonce = '{0:016X}'.format(random.getrandbits(64))
```

Comment:
This should be 128 bits (16 bytes) instead of 64 bits. The nonce is never converted to a integer, so there is no overflow problem and avoids the birthday paradox.  That is change this to:
`nonce = '{0:032X}'.format(random.getrandbits(128))`